src/runtime/js.zig

```diff
@@ -260,6 +263,24 @@ pub fn Env(comptime State: type, comptime WebApis: type) type {
             self.isolate.performMicrotasksCheckpoint();
         }
 
+        pub fn pumpMessageLoop(self: *const Self) bool {
+            if (self.platform == null) {
```

Comment:
Since these are potentially called in a tight loop, I think you can do this at comptime:

```zig
if (comptime builtin.is_test) {
    if (self.platform == null) return false;
}
// assume it's not-null in non-test.
return self.platform.?.inner.pumpMessageLoop(self.isolate, false);

Reply:
I don't understand what this is doing, so it's hard to say. 

The page's `messageLoopCallback` is not being called very often. _all_ of page.navigate (including processing all synchronous scripts) are executed before it's ever called. This could change if navigate became fully asynchronous though.

It's only being called often when we're "done" navigating and waiting for
1 - New cdp sessions
2 - I/O (i.e. an XHR request is live)

In both cases, I would think it's mostly a no-op.